In [1]:
import geopandas as gpd
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import fiona
import pyproj
import shapely.geometry as geom

Determine the obesity geometries (squares) that intersect any of the community districts   

In [120]:
# project geometries to EPSG 4326 to match the projection of the obesity map
df_cd = gpd.read_file('Community Districts/geo_export_bf9282a4-4d98-4f1a-9606-0bf283c4c69d.shp').to_crs(fiona.crs.from_epsg(4326))
df_cd

boro_cd    shape_area     shape_leng  \
0     311.0  1.031778e+08   51549.557899   
1     313.0  8.819569e+07   65821.875617   
2     312.0  9.952550e+07   52245.830495   
3     304.0  5.666322e+07   37008.100320   
4     206.0  4.266431e+07   35875.710998   
..      ...           ...            ...   
66    227.0  3.143201e+07   28391.629705   
67    401.0  1.715489e+08   90042.718108   
68    402.0  1.398915e+08   71543.044665   
69    502.0  5.931981e+08  142669.724480   
70    595.0  5.471514e+07  110237.328244   

                                             geometry  
0   POLYGON ((-73.97299 40.60881, -73.97259 40.606...  
1   POLYGON ((-73.98372 40.59582, -73.98305 40.595...  
2   POLYGON ((-73.97140 40.64826, -73.97121 40.647...  
3   POLYGON ((-73.89647 40.68234, -73.89653 40.682...  
4   POLYGON ((-73.87185 40.84376, -73.87192 40.843...  
..                                                ...  
66  POLYGON ((-73.87054 40.86967, -73.87053 40.869...  
67  MULTIPOLYGON (((-73.90647 40.79018, -73.90251 ...  
68  POLYGON ((-73.89792 40.75424, -73.89797 40.754...  
69  MULTIPOLYGON (((-74.07347 40.57839, -74.07345 ...  
70  MULTIPOLYGON (((-74.11834 40.55046, -74.11676 ...  

[71 rows x 4 columns]

In [121]:
df_obesity = gpd.read_file('NY/NY.shp').to_crs(fiona.crs.from_epsg(4326))
df_obesity 

popgte20  popbmige30  bmivsus  pctbmige30  \
0             10           4      5.1        40.0   
1              4           1     -9.9        25.0   
2              4           2     15.1        50.0   
3              4           3     40.1        75.0   
4             10           2    -14.9        20.0   
...          ...         ...      ...         ...   
644174         8           2     -9.9        25.0   
644175         5           2      5.1        40.0   
644176         5           3     25.1        60.0   
644177         4           3     40.1        75.0   
644178         4           0    -34.9         0.0   

                                                 geometry  
0       POLYGON ((-78.81671 41.99786, -78.81712 41.997...  
1       POLYGON ((-79.72667 41.99955, -79.72667 41.999...  
2       POLYGON ((-79.72461 41.99911, -79.72580 41.999...  
3       POLYGON ((-79.68400 41.99929, -79.68624 41.999...  
4       POLYGON ((-79.68175 41.99932, -79.68400 41.999...  
...                                                   ...  
644174  POLYGON ((-73.38905 45.00883, -73.39130 45.008...  
644175  POLYGON ((-73.37558 45.00883, -73.37782 45.008...  
644176  POLYGON ((-73.37333 45.00883, -73.37558 45.008...  
644177  POLYGON ((-73.37109 45.00883, -73.37333 45.008...  
644178  POLYGON ((-73.36884 45.00883, -73.37109 45.008...  

[644179 rows x 5 columns]

In [122]:
obesity_cd = gpd.sjoin(df_obesity, df_cd, how='inner', op='intersects')

In [128]:
obesity_cd

popgte20  popbmige30    bmivsus  pctbmige30  \
151889        16           5  -3.650000   31.250000   
151890        14           6   7.957143   42.857143   
151891         2           1  15.100000   50.000000   
151892        16           9  21.350000   56.250000   
151893        82          28  -0.753659   34.146341   
...          ...         ...        ...         ...   
215631       258         103   5.022481   39.922481   
215632       194          75   3.759794   38.659794   
215856       261          91  -0.034100   34.865900   
215857       354         148   6.907910   41.807910   
215858       596         263   9.227517   44.127517   

                                                 geometry  index_right  \
151889  POLYGON ((-74.24694 40.49688, -74.24919 40.496...           56   
151890  POLYGON ((-74.24470 40.49688, -74.24694 40.496...           56   
151891  POLYGON ((-74.23796 40.49688, -74.24021 40.496...           56   
151892  POLYGON ((-74.24694 40.49858, -74.24919 40.498...           56   
151893  POLYGON ((-74.24470 40.49858, -74.24694 40.498...           56   
...                                                   ...          ...   
215631  POLYGON ((-73.88986 40.90889, -73.89211 40.908...            7   
215632  POLYGON ((-73.88762 40.90889, -73.88986 40.908...            7   
215856  POLYGON ((-73.89660 40.91059, -73.89885 40.910...            7   
215857  POLYGON ((-73.89436 40.91059, -73.89660 40.910...            7   
215858  POLYGON ((-73.89211 40.91059, -73.89436 40.910...            7   

        boro_cd    shape_area     shape_leng  
151889    503.0  5.990621e+08  189000.708091  
151890    503.0  5.990621e+08  189000.708091  
151891    503.0  5.990621e+08  189000.708091  
151892    503.0  5.990621e+08  189000.708091  
151893    503.0  5.990621e+08  189000.708091  
...         ...           ...            ...  
215631    226.0  5.056641e+07   32820.398590  
215632    226.0  5.056641e+07   32820.398590  
215856    226.0  5.056641e+07   32820.398590  
215857    226.0  5.056641e+07   32820.398590  
215858    226.0  5.056641e+07   32820.398590  

[20480 rows x 9 columns]

Data Dictionary from the RTI's Obesity Map Data 

http://synthpopviewer.rti.org/obesity/download.html

Each state shapefile contains the same attribute fields. The key fields of interest are:

- `popgte20`: The population greater than or equal to 20 years old in the grid cell.
- `popbmige30`: The population whose BMI is greater than or equal to 30.0 BMI.
- `bmivsus`: The difference in the percent of the population with BMI greater than or equal to 30.0 minus the estimated national percent obese (34.9, NHANES 2011-2012). [The equation used to calculate this value is: bmivsus = (popbmige30 - 34.9)]
- `pctbmige30`: The percent of the persons in the grid who are obese. [The equation is: pctbmige30 = (popgte20 / popbmige30) * 100)]

Create buckets for Percentage Obesity based on RTI's obesity map layer http://synthpopviewer.rti.org/obesity/viewer.html

"Estimated percent of the adult population that is obese within each 250m grid cell. While individual cells may be low or high within close proximity of each other, the overall pattern of obesity is apparent from this map." 

In [131]:
def pct_obese_range(x):
    if x <= 24.24:
        return '0.0 - 24.24'
    if x > 24.24 and x <= 28.57:
        return '24.25 - 28.57'
    if x > 28.57 and x <= 32.5:
        return '28.58 - 32.5'
    if x > 32.5 and x <= 35.29:
        return '32.51 - 35.29'
    if x > 35.29 and x <= 37.5:
        return '35.30 - 37.5'
    if x > 37.5 and x <= 40.30:
        return '37.51 - 40.30'
    if x > 40.30 and x <= 43.14:
        return '40.31 - 43.14'
    if x > 43.14 and x <= 46.51:
        return '43.15 - 46.51'
    if x > 46.51 and x <= 52.38:
        return '46.52 - 52.38'
    if x > 52.38 and x <= 100.0:
        return '52.39 - 100.0'

In [133]:
obesity_cd['pct_obese_rng'] = obesity_cd['pctbmige30'].apply(pct_obese_range)
obesity_cd['pct_obese_rng'].value_counts(dropna=False)

28.58 - 32.5     2961
46.52 - 52.38    2386
32.51 - 35.29    2384
43.15 - 46.51    2239
24.25 - 28.57    2039
0.0 - 24.24      1924
37.51 - 40.30    1917
35.30 - 37.5     1886
40.31 - 43.14    1885
52.39 - 100.0     859
Name: pct_obese_rng, dtype: int64

Write obesity map by community district as a GeoJSON file

In [134]:
obesity_cd.to_file("obesity_cd.geojson", driver='GeoJSON')
